docs/dev/plugins.md

```diff
@@ -0,0 +1,161 @@
+# Developing InSpec Plugins for the v2 plugin API
+
+## Introduction
+
+### Inspiration
+
+The software design of the Inspec Plugin v2 API is deeply inspired by the Vagrant plugin v2 system.  While the Inspec Plugin v2 system is an independent implementation, acknowledgements are due to the Hashicorp team for such a well-thought-out design.
+
+### Note About versions
+
+"v2" refers to the second major version of the Plugin API.  It doesn't refer to the Inspec release number.
+
+### Design Goals
+
+* Load-on-demand. Improve `inspec` startup time by making plugins load heavy libraries only if they are being used.
+* Independent velocity. Enable passionate community members to contribute at their own pace by shifting core development into plugin development
+* Increase dogfooding. Convert internal components into plugins to reduce core complexity and allow testing in isolation
+
+### Design Anti-goals
+
+* Don't implement resources in plugins; use resource packs for that.
+
+## How Plugins are Located and Loaded
+
+### Plugins are usually gems
+
+The normal distribution and installation method is via gems, handled by the `inspec plugin` command.
+
+TODO: give basic overview of `inspec plugin` and link to docs
+
+### Plugins may also be found by path
+
+For local development or site-specific installations, you can also 'install' a plugin by path using `inspec plugin`, or edit `~/.inspec/plugins.json` directly to add a plugin.
+
+### The plugins.json file
+
+Inspec stores its list of known plugins in a file, `~/.inspec/plugins.json`. The purpose of this file is avoid having to do a gem path filesystem scan to locate plugins.  When you install, update, or uninstall a plugin using `inspec plugin`, InSpec updates this file.
```

Comment:
I'm gonna refrain from highlighting the InSpec capitalization thing after this one :smile:

Reply:
You really can just make one comment, along the lines of "Throughout this PR...."